cfn-resources/maintenance-window/cmd/resource/resource.go

```diff
@@ -204,22 +154,47 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *handlerError, nil
 	}
 
-	var res *mongodbatlas.Response
-	res, err := client.MaintenanceWindows.Reset(context.Background(), *currentModel.ProjectId)
-
+	resp, err := client.AtlasV2.MaintenanceWindowsApi.ResetMaintenanceWindow(context.Background(), *currentModel.ProjectId).Execute()
 	if err != nil {
-		_, _ = logger.Warnf("Delete - error: %+v", err)
-		return progress_events.GetFailedEventByResponse(err.Error(), res.Response), nil
+		return progress_events.GetFailedEventByResponse(err.Error(), resp), nil
 	}
 
-	// Response
-	event := handler.ProgressEvent{
+	return handler.ProgressEvent{
 		OperationStatus: handler.Success,
 		Message:         "delete successful",
-	}
-	return event, nil
+	}, nil
 }
 
 func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	return handler.ProgressEvent{}, errors.New("not implemented: List")
 }
+
+func (m Model) toAtlasModel() admin.GroupMaintenanceWindow {
+	return admin.GroupMaintenanceWindow{
+		DayOfWeek:            *m.DayOfWeek,
+		HourOfDay:            *m.HourOfDay,
+		StartASAP:            m.StartASAP,
+		AutoDeferOnceEnabled: m.AutoDeferOnceEnabled,
+	}
+}
+
+func get(client *util.MongoDBClient, currentModel Model) (*admin.GroupMaintenanceWindow, *handler.ProgressEvent) {
+	maintenanceWindow, resp, err := client.AtlasV2.MaintenanceWindowsApi.GetMaintenanceWindow(context.Background(), *currentModel.ProjectId).Execute()
```

Comment:
[not-blocking] recommendation, instead of accepting `util.MongoDBClient` accept an interface that implements `GetMaintenanceWindow`, this way mocking in the future will be easier 

Also consider making `get` accept the context